### ISSUE TITLE
Support compiling on Windows 10

### ISSUE THREAD

**Issue by ronaldfw** [RayTracing/InOneWeekend#65]
Getting the code to run on Windows 10 is fairly involved. Setting up a Visual Studio solution is straightforward but there are a number of code portability issues that come up and need some effort to resolve. 

The one that takes the most time is that drand48() does not seem to be supported on Windows 10. (I believe "MAXFLOAT" is another issue, but I would have to confirm.)

There are some suggestions to fix the drand48() issue here: https://stackoverflow.com/a/11068161 (including the option to just copy & paste the drand48() code into the project). In my own project I use <random> (C++11) to generate numbers.

**Comment by hollasch** [comment]
This is a scary road to walk down. Once this turns into a fully-buildable project, we'll need to worry about different operating systems, build systems and compilers, plus the ongoing support that goes with each of those.

(BTW, for Windows-building, a short makefile (using Visual Studio's `nmake` utility) is plenty sufficient.)

For now I'd like to work toward a more portable code base that can be plugged into a developer's platform/build-system of choice.

Specific issues that are ripe for addressing (in order of frequency):
1. Lots of mismatches between single and double precision
2. Missing `drand48()` implementation. Probably best to just provide decent portable RNG source.
3. Undeclared `M_PI` (this should be replaced with a proper C++ constant value)
4. Undeclared `MAXFLOAT`. This should use FP infinity instead of max finite value. It should get the value from `std::numeric_limits<T>::infinity`, or just define a proper constant value for this (generated as `1.0/0.0`).
5. All other compilation errors when just compiling the given source.

**Comment by ronaldfw** [comment]
Yes, totally agree and I didn't mean that build files should be provided and supported. What I would like though is that the code is portable and can be easily compiled on most relevant platforms (Linux, Mac OSX, Windows).

You already have a great list of issues and I'd be happy to help addressing them. I can start with the drand48 part, if that's ok for you.

(I have Linux, Mac OSX, and Windows 10 as platforms where I can try out compiling.)

**Comment by hollasch** [comment]
Sweet! I'll create individual issues for these.

**Comment by vchizhov** [comment]
See my fork of the repo, I believe I fixed all of the points in your first comment @hollasch
